### DISCUSSION TITLE
Weekly #3

### DISCUSSION THREAD

**Discussion by Ninetilt** [Ninetilt/Project-TDC#5] [General]
### **Übersicht**
In diesem Blog-Post zeigen wir euch Aktivitätsprogramme zu verschiedenen Anwendungsfälle unserer App, verschiedene Sequenzdiagramme zu unserem Projekt und teilen noch ein Paar Gedanken zu dem Abschluss unseres ersten Sprints.

### **Aktivitätsdiagramme**
Folgende Anwendungsfälle haben wir ausgewählt, zu denen wir Aktivitätsdiagramme erstellt haben. Die UML-Diagramme sind wie immer verlinkt.
- **[Liste erstellen](https://github.com/Ninetilt/Project-TDC/blob/main/docs/uml/PDF/CreateListActivity.pdf)**
   Das Erstellen neuer To-Do Listen von Grund auf oder mit Vorlage.
- **[Freund hinzufügen](https://github.com/Ninetilt/Project-TDC/blob/main/docs/uml/PDF/FriendRequestActivity.pdf)**
   Das Einsehen der Profile anderer Nutzer und Hinzufügen von Benutzern als Freunde.
- **[Benutzer-Login](https://github.com/Ninetilt/Project-TDC/blob/main/docs/uml/PDF/LoginActivity.pdf)**
   Das Einloggen des Benutzers in den eigenen Account.
- **[Rewarding](https://github.com/Ninetilt/Project-TDC/blob/main/docs/uml/PDF/RewardingActivity.pdf)**
   Der Prozess nach dem Abschließen einer Liste. Verdiente Punkte können für die Verbesserung und Individualisierung des Spieleravatars verwendet werden.

### **Sequenzdiagramme**
Folgende Sequenzdiagramme haben wir für unser Projekt erstellt.
- [**Freund-hinzufügen-Sequenz**](https://github.com/Ninetilt/Project-TDC/blob/main/docs/uml/PDF/AddFriendSequence.pdf)
   Der Prozess beim Hinzufügen anderer Benutzer als neue Freunde.
- [**Liste-abschließen-Sequenz**](https://github.com/Ninetilt/Project-TDC/blob/main/docs/uml/PDF/FinishListSequence.pdf)
   Der Prozess beim Abschließen einer To-Do List.

### **Abschluss Sprint 1**
Der erste Sprint ist zu Ende und dazu möchten wir ein Paar Worte verlieren. Wir haben alle gesetzten Ziele dieses Sprints erreicht. Die nötigen Entwicklungsumgebungen wurden bei allen aufgesetzt und wir haben ein klares Bild davon was unsere Ziele in diesem Projekt sind. Hauptsächlich bestand dieser Sprint aus Planug und Vorbereitung für die spätere Programmierung. 
In einem kurzen Retro haben wir uns zum Sprint ausgetauscht. Soweit sind wir mit dem ersten Sprint zufrieden, obwohl es noch nicht so viel dazu zu sagen gab. Das kommt sicherlich in den nächsten Sprints.

Falls ihr noch Fragen zu unseren Aktivitäts- oder Sequenzdiagrammen, oder zu unserem Projekt generell, schreibt es einfach in die Kommentare.

**Comment by Kartoffeleintopf** [comment]
Eure UML Diagramme erscheinen mir soweit als logisch und sinnvoll. Beim Rearding Aktivitätsdiagramm gibt es allerdings ein kleines Manko. Und zwar scheint man aus dem "Outfit kaufen"-Zustand nicht mehr raus zu kommen, wobei der eigentliche Schluss nach "Rewarding abschließen" nie erreicht wird. Euer Retrospektive bezüglich herrschen bei uns ähnliche Gefühle. Danke für diesen Einblick.
<br>
 >MFG das TripAhead Team (CN)

**Reply by Ninetilt** [reply]
Vielen Dank für die Anmerkung, da ist uns ein Fehler unterlaufen. Wir haben das Aktivitätsdiagramm entsprechend angepasst.

Viele Grüße
Projekt TDC (JS)

**Comment by Ferreidi** [comment]
Die Diagramme sehen sehr gut aus und sind verständlich. Außerdem ist es toll, wie organisiert ihr an das Projekt rangeht. Die Namen der Diagramme sind klar verständlich und auch die kleine Beschreibung darunter ist sehr gut und passend. Damit hat man schnell einen Überblick, was die einzelnen Diagramme aufzeigen. Ein kleines Problem ist im Diagramm Rewarding. Es ist zwar ein Pfeil zu Outfit kaufen aber kein Pfeil geht auch wieder zurück um wieder aufs Hauptziel zu steuern. Ist das beabsichtig oder fehlt dort einfach ein Pfeil noch? 

Wir wünschen euch noch viel Erfolg bei eurem Projekt!

Mit freundlichen Grüßen
Team Jump and Run

**Reply by Ninetilt** [reply]
Vielen Dank für die Anmerkung, bei dem Akttivitätsdiagramm fehlt ein Pfeil das ist tatsaächlich nicht beabsichtigt. WIr haben es entprechend angepasst.

Viele Grüße
Projekt TDC (JS)

**Comment by OmarMoujahed** [comment]
Die Diagramme erscheinen mir gelungen und bieten eine klare, verständliche Darstellung der verschiedenen Prozesse. Besonders positiv hervorzuheben sind die prägnanten und aussagekräftigen Titel der Diagramme sowie die kurzen Beschreibungen darunter. Diese Struktur ermöglicht es, schnell zu erfassen, welchen Teilprozess jedes Diagramm darstellt und wie die einzelnen Elemente miteinander verknüpft sind. Die Übersichtlichkeit der Diagramme zeigt zudem, dass ihr sehr organisiert an das Projekt herangeht und großen Wert auf eine gute Struktur legt, was die Nachvollziehbarkeit für alle Beteiligten erheblich verbessert.

Insgesamt vermittelt die Visualisierung eurer Arbeit einen durchdachten Eindruck und unterstützt das Verständnis der Projektstruktur und -ziele.

Wir wünschen euch weiterhin viel Erfolg bei eurem Projekt und freuen uns auf die nächsten Schritte!

Mit freundlichen Grüßen.
Project HÆLTH 🌱
